internal/handlers/vault_handler.go

```diff
@@ -16,9 +16,18 @@ func (a *Api) registerVaultHandler(c *gin.Context) {
 		return
 	}
 	// check vault already exists , should we tell front-end that vault already registered?
-	if _, err := a.s.GetVault(vault.PublicKeyECDSA, vault.PublicKeyEDDSA); err == nil {
+	if v, err := a.s.GetVault(vault.PublicKeyECDSA, vault.PublicKeyEDDSA); err == nil {
 		a.logger.Error(err)
-		_ = c.Error(errVaultAlreadyRegist)
+		c.JSON(http.StatusOK, models.VaultResponse{
+			UId:            v.Uid,
+			Name:           v.Name,
+			Alias:          v.Alias,
+			PublicKeyECDSA: v.ECDSA,
+			PublicKeyEDDSA: v.EDDSA,
+			TotalPoints:    v.TotalPoints,
+			JoinAirdrop:    v.JoinAirdrop,
+			Coins:          []models.ChainCoins{},
```

Comment:
If the vault is already registered , I prefer we don't return it to client side , because "coins" field will certainly be wrong , and it might confuse user